library/core/src/option.rs

```diff
@@ -1641,15 +1647,16 @@ impl<T> Option<T> {
     where
         F: FnOnce() -> T,
     {
-        if let None = *self {
-            // the compiler isn't smart enough to know that we are not dropping a `T`
-            // here and wants us to ensure `T` can be dropped at compile time.
-            mem::forget(mem::replace(self, Some(f())))
+        if let None = self {
+            *self = Some(f());
```

Comment:
This change makes sense to me, though -- the `forget`+`replace` is a residue from when this was unstably `const`, before that was undone.